test/integration/apiserver/tracing/tracing_test.go

```
@@ -105,93 +84,461 @@ kind: TracingConfiguration
 	)
 	defer testServer.TearDownFn()
-	clientConfig := testServer.ClientConfig
-
-	// Create a client that creates sampled traces.
-	tp := trace.TracerProvider(sdktrace.NewTracerProvider(sdktrace.WithSampler(sdktrace.AlwaysSample())))
-	clientConfig.Wrap(tracing.WrapperFor(tp))
-	clientSet, err := client.NewForConfig(clientConfig)
+	clientSet, err := client.NewForConfig(testServer.ClientConfig)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	// Make a request with the instrumented client
-	_, err = clientSet.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		t.Fatal(err)
+	for _, tc := range []struct {
+		desc          string
+		apiCall       func(*client.Clientset) error
+		expectedTrace []*spanExpectation
+	}{
+		{
+			desc: "create node",
+			apiCall: func(c *client.Clientset) error {
+				_, err = clientSet.CoreV1().Nodes().Create(context.Background(),
+					&v1.Node{ObjectMeta: metav1.ObjectMeta{Name: "fake"}}, metav1.CreateOptions{})
+				return err
+			},
+			expectedTrace: []*spanExpectation{
+				{
+					name: "KubernetesAPI",
+					attributes: map[string]func(*commonv1.AnyValue) bool{
+						"http.user_agent": func(v *commonv1.AnyValue) bool {
+							return strings.HasPrefix(v.GetStringValue(), "tracing.test")
+						},
+						"http.target": func(v *commonv1.AnyValue) bool {
+							return v.GetStringValue() == "/api/v1/nodes"
+						},
+						"http.method": func(v *commonv1.AnyValue) bool {
+							return v.GetStringValue() == "POST"
+						},
+					},
+				},
+				{
+					name: "etcdserverpb.KV/Txn",
+					attributes: map[string]func(*commonv1.AnyValue) bool{
+						"rpc.system": func(v *commonv1.AnyValue) bool {
+							return v.GetStringValue() == "grpc"
+						},
+					},
+					events: []string{"message"},
+				},
+			},
+		},
+		{
+			desc: "get node",
+			apiCall: func(c *client.Clientset) error {
+				_, err = clientSet.CoreV1().Nodes().Get(context.Background(), "fake", metav1.GetOptions{})
+				return err
```

I find this a little awkward that there is a dependency between these tests. I'm not sure if this is easy (or worth) to fix though

---

mmm... I think you are right.  If I change the name, this Get() call fails.  I could ignore the error here, but then if the test fails I can't tell if the request failing was the reason or not.  I'd probably prefer to leave it as-is.

---

That's good to know, yeah I was curious, because of how the tests are in an order that makes sense (create, get, ... delete). A different name would help, thanks!

---

OK, Feel free to leave it as is, but maybe mention with a comment that this is order-dependant.

---

added "This depends on the "create node" step having completed successfully" comments to steps that depend on creation.